File: packages/reports/addon/models/bard-request/fragments/sort.js
/**
 * Copyright 2017, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import DS from 'ember-data';
import MF from 'model-fragments';
import { validator, buildValidations } from 'ember-cp-validations';

const { Fragment } = MF;

const Validations = buildValidations({
  metric: validator('presence', {
    presence: true,
    message: 'The metric field in sort cannot be empty'
  }),
  direction: validator('presence', {
    presence: true,
    message: 'The direction field in sort cannot be empty'
  })
});

export default Fragment.extend(Validations, {
  metric:         DS.attr('sort'),
  direction:      DS.attr('string', { defaultValue: 'desc'})
});

File: packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/fragments/request/v2/Sort.kt
/**
 * Copyright 2020, Yahoo Holdings Inc.
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */
package com.yahoo.navi.ws.models.beans.fragments.request.v2

import com.yahoo.navi.ws.models.beans.enums.ColumnType
import org.hibernate.annotations.Parameter
import org.hibernate.annotations.Type

data class Sort(
    var field: String,
    @get: Type(type = "com.yahoo.navi.ws.models.types.JsonType", parameters = [
        Parameter(name = "class", value = "kotlin.collections.HashMap")
    ]) var parameters: Map<String, String>,
    var type: ColumnType?,
    var direction: String
) {
    constructor() : this(
            "",
            emptyMap(),
            null,
            ""
    )
}